src/frontend/Debug_data_generation.ml

```diff
@@ -104,26 +129,84 @@ let gen_vector m n t =
       wrap_vector (List.map ~f:wrap_real l)
   | _ -> {int_two with expr= PostfixOp (gen_row_vector m n t, Transpose)}
 
-let gen_identity_matrix n m =
-  { int_two with
-    expr=
-      RowVectorExpr
-        (List.map
-           (List.range 1 (n + 1))
-           ~f:(fun k ->
-             wrap_row_vector
-               (List.map ~f:wrap_real
-                  ( repeat (min (k - 1) m) 0.
-                  @ (if k <= m then [1.0] else [])
-                  @ repeat (m - k) 0. )) )) }
-
-let gen_matrix mm n m t =
+let gen_cov_unwrapped n =
+  let l = repeat_th (n * n) (fun _ -> Random.float 2.) in
+  let l_mat = vect_to_mat l n in
+  matprod l_mat (transpose l_mat)
+
+let wrap_real_mat m =
+  let mat_wrapped =
+    List.map ~f:wrap_row_vector
+      (List.map ~f:(fun x -> List.map ~f:wrap_real x) m)
+  in
+  {int_two with expr= RowVectorExpr mat_wrapped}
+
+let gen_diag_mat l =
+  let n = List.length l in
+  List.map
+    (List.range 1 (n + 1))
+    ~f:(fun k ->
+      repeat (min (k - 1) n) 0.
+      @ (if k <= n then [List.nth_exn l (k - 1)] else [])
+      @ repeat (n - k) 0. )
+
+let fill_lower_triangular m =
+  let fill_row i l =
+    let _, tl = List.split_n l i in
+    List.init ~f:(fun _ -> Random.float 2.) i @ tl
+  in
+  List.mapi ~f:fill_row m
+
+let pad_mat mm m n =
+  let padding_mat =
+    List.init (m - n) ~f:(fun _ -> List.init n ~f:(fun _ -> 0.))
```

Comment:
Does this need to be all zeros?

Reply:
No, I did this to be consistent with the previous behaviour (zeros padding the identity matrix), but it's probably better to put random numbers instead. Changed.